src/transformers/models/bark/configuration_bark.py

```diff
@@ -301,7 +301,8 @@ def __init__(
         self.semantic_config = BarkSemanticConfig(**semantic_config)
         self.coarse_acoustics_config = BarkCoarseConfig(**coarse_acoustics_config)
         self.fine_acoustics_config = BarkFineConfig(**fine_acoustics_config)
-        self.codec_config = AutoConfig.for_model(**codec_config)
+        codec_model_type = codec_config["model_type"] if "model_type" in codec_config else "encodec"
+        self.codec_config = CONFIG_MAPPING[codec_model_type](**codec_config)
```

Comment:
We need to provide a default model type. This is similar to `Blip2Config`.

`AutoConfig.for_model(**codec_config)` will fail if no `model_type` is in it.